llvm-external-projects/iree-dialects/lib/Transforms/TransformMatchers.cpp

```diff
@@ -335,6 +335,20 @@ transform_ext::StructuredOpMatcher::dim(AllDims tag, CaptureDims captures) {
   return *this;
 }
 
+transform_ext::StructuredOpMatcher &
+transform_ext::StructuredOpMatcher::convolutionDims(CaptureConvDims convDims) {
+  predicates.push_back([=](linalg::LinalgOp linalgOp) -> bool {
+    LLVM_DEBUG(DBGS() << "capture convolution dimensions");
+    if (!linalg::detail::getMatchConvolutionMessage(
+             mlir::linalg::detail::isConvolutionInterfaceImpl(linalgOp,
+                                                              &convDims.value))
+             .empty())
+      return false;
+    return true;
```

Comment:
Nit: you can do directly `return linalg::detail::getMatchConvolutionMessage(...).empty()` instead of a branch. Though it may be better to actually send the message to the debug stream when it's non-empty.